### PR TITLE
Release 3.5.0

### PR DESCRIPTION
**Pull request by andrew-platt**
**Feature or improvement description**
Pull request to merge `rc-3.5.0` into `main` and create a tagged release for v3.5.0.

See the milestone and project pages for additional information

- https://github.com/OpenFAST/openfast/milestone/9
- https://github.com/orgs/OpenFAST/projects/1

Test results, if applicable
See GitHub Actions

### Release checklist:
- [x] Update the documentation version in docs/conf.py
- [x] Update the versions in docs/source/user/api_change.rst
- [x] Verify readthedocs builds correctly
- [ ] Create a tag in OpenFAST
- [x] Create a merge commit in r-test and add a corresponding tag
- [ ] Compile executables for Windows builds
    - [ ] FAST_SFunc.mexw64
    - [ ] OpenFAST-Simulink_x64.dll
    - [ ] openfast_x64.exe
    - [ ] DISCON.dll (x64)
    - [ ] AeroDyn_Driver
    - [ ] AeroDyn_Inflow_C_Binding
    - [ ] BeamDyn_Driver
    - [ ] HydroDyn_Driver
    - [ ] HydroDyn_C_Binding (x64)
    - [ ] InflowWind_Driver
    - [ ] IfW_C_Binding (x64)
    - [ ] MoorDyn_Driver
    - [ ] FAST.Farm (x64)

# Changelog

### Build systems

#1010 Use Cmake OBJECT libraries to create openfastlib and add option to use downloaded reference lapack and blas sources (`USE_LOCAL_STATIC_LAPACK` option in CMake)   @reos-rcozier
#1427 Improve CMake's `FindMKL` to avoid rutime issues and preloading libraries   @gbarter
#1506 Fix compilation issue, using Intel OneAPI on Linux (missing `-fPIC` flag)   @pablo-benito
#1530 Add missing objects from #1010
#1535 [BugFix] CMake build on Windows and Ninja build on all platforms
#1538, #1542 Add support for Flang (Classic) compiler
#1555, #1566 Modernize CMake, remove object libraries, build Simulink MEX
#1561 [BugFix] typo in souce file list for Visual Studio build of AeroDyn driver   @bjonkman

### Documentation

#1419 update manual regression testing syntax for Windows
#1559 [BugFix] Limit urllib3 version so that readthedocs builds

### OpenFAST

#1453, #1560 Remove QuKi (quad-precision) from OpenFAST

### FAST.Farm

#1504 Increase number of VTK output planes from 9 to 99

### TurbSim

#1550 [BugFix] fix index into Z array for text grid files    @bjonkman

### AeroDyn

#1477 Allow tower, hub, nacelle, and rotor below platform for MHK turbines
#1509 Improved coning and prebend modeling in future BEM
#1541 AeroDyn_Inflow minor updates
#1567 fix some divide by zero issues

### BeamDyn

#1474 BeamDyn driver visualization, and driver bugfix
#1482 Fix convergence issue at initialization

### ElastoDyn

#1477 Allow tower, hub, nacelle, and rotor below platform for MHK turbines.

### HydroDyn

#1480 [BugFix] NBody>1 && NBodyMod==1 bug in WAMIT2

### InflowWind

#1464 Add LIDAR simulation within InflowWind with control channels passed to controller   @Russell9798
#1534 [BugFix] potential error when requesting wind below the ground (also fix build issue with SD syntax)   @bjonkman
#1516, #1543, #1549 InflowWind restructure

### OpenFOAM

#1493 Allow Non-Uniform Force Point Distribution on Blades   @mchurchf

## API changes

#1493 C++ interface for OpenFOAM module includes `NodeClusterType_c` argument

## Input file changes

#1464 New section in InflowWind input file for LIDAR inputs

Full list of changes: https://openfast.readthedocs.io/en/main/source/user/api_change.html

Full input file sets: https://github.com/OpenFAST/r-test/tree/v3.5.0 (example input files from the regression testing)

### CODE REVIEW

File: docs/source/user/inflowwind/driver.rst [hunk 1]
```diff
@@ -137,4 +138,100 @@ When converting from a full-field wind format to a uniform wind file, the follow
...
-  2. Calculated by using the mean wind speeds at two points: the reference (hub) height and the uppermost height on the grid.
\ No newline at end of file
+  2. Calculated by using the mean wind speeds at two points: the reference (hub) height and the uppermost height on the grid.
+
+accel flag
+-------------------
+
+The ability to calculate the acceleration of the flow field was added to InflowWind
+to support the analysis of MHK, underwater, turbines. The acceleration is needed
+to calculate the mass effects of the fluid interacting with the rotor. Calculation of the
```
Comment (jjonkman) [inline docs/source/user/inflowwind/driver.rst:148]: Change "mass effects" to "inertia effects" or "fluid-inertia effects.

File: docs/source/user/inflowwind/driver.rst [hunk 2]
```diff
@@ -137,4 +138,100 @@ When converting from a full-field wind format to a uniform wind file, the follow
...
+acceleration is supported for Uniform/Steady Wind and grid based wind profiles (Turbsim,
+HAWC, and Bladed files). Enabling this flag causes the driver to output the flow field
+acceleration for points defined in the Points file in addition to the velocities at those
+same points.
+
+
+BoxExceedAllow flag
+-------------------
+
+A feature was added to InflowWind to all for some requested points to lie
```
Comment (jjonkman) [inline docs/source/user/inflowwind/driver.rst:158]: Wording?  "to all for some..."

**Review by jjonkman — APPROVED** [review]
Looks good to me; I'm only recommending two small documentation changes.  Thanks!

File: docs/source/user/inflowwind/driver.rst (hunk 1 above)
Comment (andrew-platt) [reply docs/source/user/inflowwind/driver.rst]: Done!

File: docs/source/user/inflowwind/driver.rst (hunk 2 above)
Comment (andrew-platt) [reply docs/source/user/inflowwind/driver.rst]: Done!

Comment (jjonkman) [reply docs/source/user/inflowwind/driver.rst]: Thanks!

**Review by deslaughter — APPROVED** [review]

**Pull request merged**